axelor-human-resource/src/main/java/com/axelor/apps/hr/web/timesheet/TimesheetLineController.java

```diff
@@ -142,4 +147,35 @@ public void setProduct(ActionRequest request, ActionResponse response) {
     response.setAttr(
         "product", "value", Beans.get(TimesheetLineService.class).getDefaultProduct(timesheetLine));
   }
+
+  public void setTimesheet(ActionRequest request, ActionResponse response) {
+    try {
+      TimesheetLine timesheetLine = request.getContext().asType(TimesheetLine.class);
+      timesheetLine = Beans.get(TimesheetCreateService.class).getOrCreateTimesheet(timesheetLine);
+      response.setValues(timesheetLine);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void setTimesheetDomain(ActionRequest request, ActionResponse response) {
+    try {
+      TimesheetLine timesheetLine = request.getContext().asType(TimesheetLine.class);
+      String idList = "0";
+      if (timesheetLine == null) {
+        response.setAttr("timesheet", "domain", "self.id IN (" + idList + ")");
+        return;
+      }
+
+      List<Timesheet> timesheetList =
+          Beans.get(TimesheetDomainService.class).getTimesheetQuery(timesheetLine).fetch();
+      if (!ObjectUtils.isEmpty(timesheetList)) {
+        idList = StringHelper.getIdListString(timesheetList);
+      }
```

Comment:
this method manages the case where the list is empty or null, so you can remove the check.